pkg/crc/machine/stop.go

```diff
@@ -9,6 +11,11 @@ import (
 )
 
 func (client *client) Stop() (state.State, error) {
+	if err := cleanKubeconfig(getGlobalKubeConfigPath(), getGlobalKubeConfigPath()); err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			logging.Warnf("Failed to remove crc contexts from kubeconfig: %v", err)
+		}
+	}
```

Comment:
@rohanKanojia this block of code should be at the end, we shouldn't clean the kubeconfig until stop the instance first.

Reply:
I had moved this above in order to make it easier to test. I had added tests in kubeconfig_test to verify that calling cleanupKubeconfig multiple times wouldn't affect kubeconfig.

Let me try to adapt tests after moving it to the bottom.

Reply:
Hello, on second thought from the user's perspective, we should always clean up `.kube/config` regardless of whether the instance was stopped successfully or not (to avoid an inconsistent state, which was the problem user was facing).

Do you think it would be okay if we move this cleanup statement in a `defer` block ?
```go
	defer func(input, output string) {
		err := cleanKubeconfig(input, output)
		if !errors.Is(err, os.ErrNotExist) {
			logging.Warnf("Failed to remove crc contexts from kubeconfig: %v", err)
		}
	}(getGlobalKubeConfigPath(), getGlobalKubeConfigPath())
```

Reply:
yes, having it part of defer would be also good so that it is execute regardless.